src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -3538,7 +3538,7 @@ export function getFilteredMolecularProfilesByAlterationType(
                         profile.molecularAlterationType === alterationType;
                     if (!_.isEmpty(allowedDataTypes)) {
                         isFiltered =
-                            isFiltered ||
```

Comment:
Nice catch. I guess we have been ignoring allowed data types as long as alteration type matches.

Reply:
Yes, I think so. We should try to include unit tests for these general functions so that when we need to modify it next time, we can modify it with confidence.